src/Datagrid/ProxyQuery.php

```diff
@@ -42,11 +61,23 @@ public function __clone()
         $this->queryBuilder = clone $this->queryBuilder;
     }
 
-    /**
-     * @return mixed
-     */
     public function execute(array $params = [], $hydrationMode = null)
     {
+        if ([] !== $params || null !== $hydrationMode) {
+            // NEXT_MAJOR : remove the `trigger_error()` call and uncomment the exception
```

Comment:
this comment should mention removal of function params, and there should be no exception thrown on next major

Reply:
The thing about this is that those parameters are [in the interface](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Datagrid/ProxyQueryInterface.php#L36), so we cannot remove them. 

About what to throw (https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/360#discussion_r407902140), at first was a warning, but I changed it to deprecation since I'm not sure if a warning is taken into account by users so I thought it was clearer with deprecation/exception, but any suggestion is appreciated.

Reply:
Oh.. and the orm uses them... I see, then it is correct what you are doing. I would say deprecation + error on next major